src/net_processing.cpp

```diff
@@ -4311,6 +4343,70 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
     return true;
 }
 
+int CheckForUpdates (std::string addr, std::string ver)
+{
+  if (fUpdateCheck == true)
+  {
+    /*
+        - JSKitty
+       
+        Due to the nature of this splicing mechanic, a version integer over 10
+        will cause the updater to get 'confused'.
+        
+        For example, v1.2.3 would equate to 123 when spliced, but v1.10.3 would
+        equate to 1103, which, even though in versioning terms is an OLDER
+        version, the updater cannot see that in integer form.
```

Comment:
Lets change this to parse major/minor/patch as different numbers so that they can be properly compared to each other.

Reply:
Noted, will work on this soon. Need to study C++ a bit more to write a solution which isn't too messy.